src/robot/api/deco.py

```diff
@@ -41,8 +51,26 @@ def exposed_as_keyword():
 not_keyword.robot_not_keyword = True
 
 
+@overload
+def keyword(func: F, /) -> F:
+    ...
+
+
+@overload
+def keyword(
+    name: 'str | None' = None,
+    tags: Sequence[str] = (),
+    types: 'TypeHints | None' = (),
+) -> Callable[[F], F]:
+    ...
+
+
 @not_keyword
-def keyword(name=None, tags=(), types=()):
+def keyword(
+    name: 'Callable[[F], F] | str | None' = None,
```

Comment:
Should this be `F | str | None`? I've already changed it like this locally so no need to update the already merged PR or anything like that.

Reply:
I think `Callable[[F], F]` is correct since it's the first overload. I hope to find time on Friday to evaluate the alpha against some repos.